core/src/main/scala/kafka/coordinator/group/GroupMetadata.scala

```diff
@@ -244,6 +244,25 @@ private[group] class GroupMetadata(val groupId: String, initialState: GroupState
       leaderId = members.keys.headOption
   }
 
+  def maybeElectNewLeader() {
```

Comment:
Perhaps the name should convey the fact that we are choosing from the members which are awaiting join?

Reply:
Perhaps `maybeElectJoinedLeader`? Another question I'm wondering is if it's possible to reach this state when _all_ the static members are offline?